File: backend/app/schemas/message.py
from pydantic import BaseModel
from datetime import datetime

class CreateMessageRequest(BaseModel):
    session_id: str
    role: str
    content: str
    
class MessageResponse(BaseModel):
    id: str
    session_id: str
    role: str
    content: str
    created_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend/lib/api/types/message.ts
export type MessageRole = "user" | "assistant";


export interface CreateMessageRequest {
  session_id: string;
  role: string;
  content: string;
}

export interface MessageResponse {
  id: string;
  session_id: string;
  role: MessageRole;
  content: string;
  created_at: string;
}